pkg/testhelper/accessory.go

```diff
@@ -268,16 +261,45 @@ func GetFreePort(t TestingTInterface) string {
 		if err != nil {
 			t.Fatalf("could not listen on free port: %v", err)
 		}
-		defer func() {
-			if err := l.Close(); err != nil {
+		defer func(c io.Closer) {
+			if err := c.Close(); err != nil {
 				t.Errorf("could not close listener: %v", err)
 			}
-		}()
+		}(l)
 		port := strconv.Itoa(l.Addr().(*net.TCPAddr).Port)
-		if _, previouslyAllocated := ports.LoadOrStore(port, nil); !previouslyAllocated {
+		// Tests run in -parallel will run in separate processes, so we must use the file-system
+		// for sharing state and locking across them to coordinate who gets which port. Without
+		// some mechanism for sharing state, the following race is possible:
+		// - process A calls net.ListenTCP() to resolve a new port
```

Comment:
calling `ListenTCP` here is now pointless

Reply:
.... why?